tests/webapp/api/test_performance_data_api.py

```diff
@@ -235,3 +236,115 @@ def test_filter_data_by_framework(webapp, test_repository, test_perf_signature,
     assert len(datums) == 1
     assert datums[0]['job_id'] == 1
     assert datums[0]['signature_id'] == 2
+
+
+def test_filter_signatures_by_interval(webapp, test_repository, test_perf_signature,
+                                        test_perf_signature_same_hash_different_framework):
+    # interval for the last 24 hours, only one signature exists last updated within that timeframe
+    resp = webapp.get(reverse('performance-signatures-list',
+                              kwargs={
+                                  "project": test_perf_signature.repository.name
+                              }) + '?interval={}'.format(86400))
+    assert resp.status_int == 200
+    assert len(resp.json.keys()) == 1
+    assert resp.json[test_perf_signature.signature_hash]['id'] == 2
+
+
+def test_filter_signatures_by_range(webapp, test_repository, test_perf_signature,
+                                        test_perf_signature_same_hash_different_framework):
+    # start date now, interval last 24 hours
+    now_epoch = int(time.time())
+    client = APIClient()
+    resp = webapp.get(reverse('performance-signatures-list',
+                              kwargs={
+                                  "project": test_perf_signature.repository.name
+                              }) + '?interval={}&start_date={}'.format(86400, now_epoch))
```

Comment:
I don't think this is going to work because `now_epoch` is going to be newer than the date of the signature.

I'd probably redesign the API as described below (which will make errors like this easier to spot) and reworking the test to get a range from "one day ago" -> "now".